Domain/Submission/src/main/java/tw/waterball/judgegirl/submission/domain/usecases/SubmitCodeUseCase.java

```diff
@@ -56,11 +56,24 @@ public void execute(SubmitCodeRequest request, SubmissionPresenter presenter) th
 
         Submission submission = saveSubmissionWithCodes(submission(request), request.fileResources);
 
-        mayDeployJudgerIfNotJudged(request, problem, submission)
+        mayDeployJudgerAndPublishEvent(problem, submission);
+
+        presenter.setSubmission(submission);
+    }
+
+    public void execute(Submission originalSubmission) throws SubmissionThrottlingException {
```

Comment:
Rename `originalSubmission` to `submission`
 `SubmitCodeUseCase` assumes that the incoming `submission` should be a new one which doesn't have ID and the verdict. (So you need to clear them in the rejudge usecase.)